### PR TITLE
Deploy update

### PR DESCRIPTION
**Pull request by nharper285**
## Summary of the Pull Request

_What is this about?_
This PR removes Python App Function deployment code and `enable_dotnet` code. It is the first of several PRs that will remove references to the old python app service. 

For now, we are still deploying two app functions, but they will _both_ be dotnet functions. Any code that can be shared (in `deploy.py` or in any of the `*.bicep` files) is now shared and old python-specific references and resources have been removed.

This PR also updates the names of the zips output in the release artifacts. The dotnet api zip is now `api-service.zip`, i.e. the primary zip. 

## PR Checklist
* [ ] Applies to work item: #2635 
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

_What does this include?_

## Validation Steps Performed

_How does someone test & validate?_

### CODE REVIEW

**Review by chkeita — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -205,7 +205,7 @@ jobs:
         set -ex
         cd contrib/deploy-onefuzz-via-azure-devops
         python -m pip install --upgrade pip
-        python -m pip install tox pipenv
+        python -m pip install tox pipenv==2022.11.11
```
Comment (mgreisen) [inline .github/workflows/ci.yml:208]: Can you create a work item to un-pin this once the root cause is fixed?

Comment (nharper285) [reply .github/workflows/ci.yml:208]: https://github.com/microsoft/onefuzz/issues/2651

File: src/deployment/azuredeploy.bicep [hunk 2]
```diff
@@ -80,29 +75,16 @@ module operationalInsights 'bicep-templates/operational-insights.bicep' = {
...
-module linuxServerFarm 'bicep-templates/server-farms.bicep' = {
-  name: 'linux-server-farm'
+module serverFarm 'bicep-templates/server-farms.bicep' = {
+  name: 'server-farm'
   params: {
     server_farm_name: name
     owner: owner
     location: location
-    use_windows: false
-    create: true
```
Comment (mgreisen) [inline src/deployment/azuredeploy.bicep:90]: I know you've tested, but we don't need the `create : true` property?

Comment (nharper285) [reply src/deployment/azuredeploy.bicep:90]: Nope!

**Review by mgreisen — APPROVED** [review]

**Review by AdamL-Microsoft — APPROVED** [review]

**Pull request merged**